pos_order_return/models/pos_order.py

```diff
@@ -0,0 +1,193 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016-2018 Sylvain LE GAL (https://twitter.com/legalsylvain)
+# Copyright 2018 David Vidal <david.vidal@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+from odoo import _, api, fields, models
+from odoo.addons import decimal_precision as dp
+from odoo.exceptions import ValidationError
+
+
+class PosOrder(models.Model):
+    _inherit = 'pos.order'
+
+    returned_order_id = fields.Many2one(
+        comodel_name='pos.order',
+        string='Returned Order',
+        readonly=True,
+    )
+    refund_order_ids = fields.One2many(
+        comodel_name='pos.order',
+        inverse_name='returned_order_id',
+        string='Refund Orders',
+        readonly=True,
+    )
+    refund_order_qty = fields.Integer(
+        compute='_compute_refund_order_qty',
+        string='Refund Orders Quantity',
+        digits=dp.get_precision('Product Unit of Measure'),
+    )
+
+    def _compute_refund_order_qty(self):
+        for order in self:
+            order.refund_order_qty = len(order.refund_order_ids)
+
+    def _blank_refund(self, res):
+        self.ensure_one()
+        new_order = self.browse(res['res_id'])
+        new_order.returned_order_id = self
+        # Remove created lines and recreate and link Lines
+        new_order.lines.unlink()
+        return new_order
+
+    def _prepare_invoice(self):
+        res = super(PosOrder, self)._prepare_invoice()
+        if not self.returned_order_id:
+            return res
+        res.update({
+            'origin': self.name,
+            'type': 'out_refund',
+            'refund_invoice_id': self.returned_order_id.invoice_id.id,
+        })
+        return res
+
+    def _action_create_invoice_line(self, line=False, invoice_id=False):
+        line = super(PosOrder, self
+                     )._action_create_invoice_line(line, invoice_id)
+        if not self.returned_order_id:
+            return line
+        # Goes to refund invoice thus it should be positive
+        line.quantity = -line.quantity
+        return line
+
+    def _action_pos_order_invoice(self):
+        """Wrap common process"""
+        self.action_pos_order_invoice()
+        self.invoice_id.sudo().action_invoice_open()
+        self.account_move = self.invoice_id.move_id
```

Comment:
Why this here in the return?

Reply:
I'm replicating the same behaviour that `point_of_sale` has in normal mode

Reply:
@pedrobaeza : are you ok with the answer of @chienandalu ?

Reply:
I have to check with him again as I don't understand the reasons for invoicing the return.

Reply:
It comes from here: https://github.com/OCA/pos/pull/269/commits/00db862fdf891182563ef8a1342347e939da9075#diff-3735e1293bd7b48ab8f6f80fac232206R107 If the original order had an invoice, a refund invoice is made automatically

Reply:
OK, got it.